### PR TITLE
Adapt storage UI

### PR DESCRIPTION
**Pull request by imobachgs**
Adapt the Storage components to the new Storage D-Bus API.

### CODE REVIEW

File: web/src/Proposal.jsx [hunk 1]
```diff
@@ -1,26 +1,15 @@
 import React from "react";
```
Comment (dgdavid) [inline web/src/Proposal.jsx:1]: NP: I'd rename this component to ProposalActions or similar.

File: web/src/Proposal.jsx [hunk 2]
```diff
@@ -1,26 +1,15 @@
...
 import {
-  TableComposable,
-  Thead,
-  Tr,
-  Th,
-  Tbody,
-  Td
-} from "@patternfly/react-table";
-
-import filesize from "filesize";
```
Comment (dgdavid) [inline web/src/Proposal.jsx:11]: Remember to remove the _filesize_ dependency, no longer needed.

File: web/src/Storage.jsx [hunk 3]
```diff
@@ -31,47 +33,57 @@ export default function Storage() {
...
-      const [_, changes] = args;
-      if (Object.keys(changes).includes("Disk")) {
-        client
-          .getStorage()
-          .then(proposal =>
-            dispatch({ type: "UPDATE_PROPOSAL", payload: proposal })
-          );
+      const [iface, properties] = args;
+
+      if (iface !== "org.opensuse.DInstaller.Storage.Actions1") {
```
Comment (dgdavid) [inline web/src/Storage.jsx:63]: Maybe we could have a JS module maping ifaces and exporting them as constants. I mean, a _ifaces.js_ file or so

```js
const ACTIONS = "org.opensuse.DInstaller.Storage.Actions1";
const  WHATERVER = "org.opensuse.DInstaller.Storage.Actions1";

...

export {
  ACTIONS,
  WHATEVER,
  ...
};
```

Then, 

```js

import { ACTIONS as ACTIONS_IFACE } from 'ifaces';

...

if (iface !== ACTIONS_IFACE) {
 
...

```

File: web/src/Storage.jsx [hunk 4]
```diff
@@ -31,47 +33,57 @@ export default function Storage() {
...
+        return;
       }
+
+      const newActions = properties.All.v.map(action => {
+        const { Text: textVar, Subvol: subvolVar } = action.v;
+        return { text: textVar.v, subvol: subvolVar.v };
+      });
```
Comment (dgdavid) [inline web/src/Storage.jsx:70]: I'd move that mapping to an util function.

File: web/src/lib/InstallerClient.test.js [hunk 5]
```diff
@@ -10,16 +10,14 @@ const DBUS_PATH = "/org/opensuse/YaST/Installer";
 const DBUS_IFACE = "org.opensuse.YaST.Installer";
 const LANGUAGE_IFACE = "org.opensuse.DInstaller.Language1";
 const SOFTWARE_IFACE = "org.opensuse.DInstaller.Software1";
+const STORAGE_PROPOSAL_IFACE = "org.opensuse.DInstaller.Storage.Proposal1";
+const STORAGE_ACTIONS_IFACE = "org.opensuse.DInstaller.Storage.Actions1";
```
Comment (dgdavid) [inline web/src/lib/InstallerClient.test.js:14]: They could be in the proposed _ifaces.js_ module.

**Review by dgdavid — APPROVED** [review]
Just minor suggestions, but looks good.

**Comment by joseivanlopez** [comment]
LGTM for the backend part. For now, I think it is enough. Let's see if we need something more complex in the future.

File: web/src/Storage.jsx (hunk 4 above)
Comment (imobachgs) [reply web/src/Storage.jsx:70]: I prefer to wait until we improve `InstallerClient`. In the component, you shouldn't need to deal with D-Bus variants.

File: web/src/Storage.jsx (hunk 3 above)
Comment (imobachgs) [reply web/src/Storage.jsx:63]: The idea is that this callback does not expose D-Bus internals (so you should not get the interface name and so on). Let's wait until we refactor `InstallerClient`.

File: web/src/Proposal.jsx (hunk 2 above)
Comment (imobachgs) [reply web/src/Proposal.jsx:11]: Good point.

File: web/src/Proposal.jsx (hunk 1 above)
Comment (imobachgs) [reply web/src/Proposal.jsx:1]: Yes. And I would change the wording about storage in general: e.g., replacing "Target" with "Partitioning" or something like that. Let's do that in an open PR.

File: web/src/lib/InstallerClient.test.js (hunk 5 above)
Comment (imobachgs) [reply web/src/lib/InstallerClient.test.js:14]: Something for the `InstallerClient` refactoring.

**Pull request merged**